tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -4406,6 +4405,202 @@ TEST_F(InputProcessorFixture, epJSONgetObjectItem_minfields)
+        "    DummySch,                !- Use Flow Rate Fraction Schedule Name",
+        "    ,                        !- Cold Water Supply Temperature Schedule Name",
+        "    ,                        !- Use Side Inlet Node Name",
+        "    ,                        !- Use Side Outlet Node Name",
+        "    ,                        !- Use Side Effectiveness",
+        "    Zone4WaterOutletNode,    !- Source Side Inlet Node Name",
+        "    Zone4WaterInletNode,     !- Source Side Outlet Node Name",
+        "    0.9;                     !- Source Side Effectiveness",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    state->dataGlobal->isEpJSON = false;
+    state->dataInputProcessing->inputProcessor->initializeMaps();
+    std::string alphaFieldValue;
+    Real64 numericFieldValue = 0.0;
+    json objectSchemaProps;
+
+    // Water heater object
+    std::string obj_type1 = "WaterHeater:Mixed";
+    objectSchemaProps = state->dataInputProcessing->inputProcessor->getObjectSchemaProps(*state, obj_type1);
+    auto instances = state->dataInputProcessing->inputProcessor->epJSON.find(obj_type1);
+    if (instances != state->dataInputProcessing->inputProcessor->epJSON.end()) {
+        // globalSolverObject.referenceConditions.clear();
+        auto &instancesValue = instances.value();
+        for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+            auto const &fields = instance.value();
+            // User inputs from above
+            // An autosized field will return -99999
+            numericFieldValue =
+                state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type1, fields, objectSchemaProps, "tank_volume");
+            EXPECT_EQ(numericFieldValue, -99999);
+            numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(
+                *state, obj_type1, fields, objectSchemaProps, "heater_maximum_capacity");
+            EXPECT_EQ(numericFieldValue, -99999);
+            // Check a numeric field that's not autosized
+            numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(
+                *state, obj_type1, fields, objectSchemaProps, "source_side_effectiveness");
+            EXPECT_EQ(numericFieldValue, 0.9);
+            // Also check a field that defaults to autosize (not input above)
+            numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(
+                *state, obj_type1, fields, objectSchemaProps, "use_side_design_flow_rate");
+            EXPECT_EQ(numericFieldValue, -99999);
+            // Check an alpha field
+            alphaFieldValue = state->dataInputProcessing->inputProcessor->getAlphaFieldValue(
+                *state, obj_type1, fields, objectSchemaProps, "on_cycle_parasitic_fuel_type");
+            EXPECT_TRUE(UtilityRoutines::SameString(alphaFieldValue, "Electricity"));
```

Comment:
Interesting, this gets pushed to lowercase since the field is an enum choice.

    "on_cycle_parasitic_fuel_type": {
        "type": "string",
        "enum": [
            "Electricity",